edelphi/src/main/java/fi/metatavu/edelphi/utils/ReportUtils.java

```diff
@@ -318,177 +318,154 @@ public static void zipCharts(RequestContext requestContext, OutputStream outputS
+        if (chartFile != null) {
+          Logging.logInfo("SVG file from " + svgUri + " uploaded into Google Drive with id: " + chartFile.getId());
+        } else {
+          Logging.logInfo("Uploading failed to Google Drive for SVG file: " + svgUri);
         }
-        connection.disconnect();
       }
+    }
 
-      // .. tidy it a bit
-
-      ByteArrayOutputStream tidyXHtml = new ByteArrayOutputStream();
-      Tidy tidy = new Tidy();
-      tidy.setInputEncoding("UTF-8");
-      tidy.setOutputEncoding("UTF-8");
-      tidy.setShowWarnings(true);
-      tidy.setNumEntities(false);
-      tidy.setXmlOut(true);
-      tidy.setXHTML(true);
-      tidy.setWraplen(0);
-      tidy.setQuoteNbsp(false);
-      tidy.parse(new StringReader(reportHtml), tidyXHtml);
-
-      DocumentBuilderFactory builderFactory = DocumentBuilderFactory.newInstance();
-      builderFactory.setNamespaceAware(false);
-      builderFactory.setValidating(false);
-      builderFactory.setFeature("http://xml.org/sax/features/namespaces", false);
-      builderFactory.setFeature("http://xml.org/sax/features/validation", false);
-      builderFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-dtd-grammar", false);
-      builderFactory.setFeature("http://apache.org/xml/features/nonvalidating/load-external-dtd", false);
-      DocumentBuilder builder = builderFactory.newDocumentBuilder();
-
-      ByteArrayInputStream inputStream = new ByteArrayInputStream(tidyXHtml.toByteArray());
-
-      // Tidied document needs to be parsed into a document, so we can do some
-      // changes into it
-      Document reportDocument = builder.parse(inputStream);
-
-      // Google Drive can not access our style sheets, so we need to embed them
-      // directly into the document
-      NodeList linkList = XPathAPI.selectNodeList(reportDocument, "//link");
-      for (int i = 0, l = linkList.getLength(); i < l; i++) {
-        Element linkElement = (Element) linkList.item(i);
-        String linkRel = linkElement.getAttribute("rel");
-        if (StringUtils.equalsIgnoreCase(linkRel, "stylesheet")) {
-          String href = linkElement.getAttribute("href");
-          Logging.logInfo("Embedding css from " + href + " into Google report");
-          String cssText = CSSUtils.downloadCSS(hostUrl + href, true).replaceAll("[\n\r]", " ");
-
-          Node parent = linkElement.getParentNode();
-          Element styleElement = reportDocument.createElement("style");
-          styleElement.setAttribute("type", "text/css");
-          styleElement.appendChild(reportDocument.createTextNode(cssText));
-
-          parent.replaceChild(styleElement, linkElement);
-        }
-      }
+    return targetFolder.getId();
+  }
+  
+  public static String uploadReportToGoogleDrive(Locale locale, Drive drive, URL url, String queryName, int retryCount) throws IOException, TransformerException, ParserConfigurationException, SAXException {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Refactor this method to throw at most one checked exception instead of: java.io.IOException, javax.xml.transform.TransformerException, javax.xml.parsers.ParserConfigurationException, org.xml.sax.SAXException [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS1160)